File: HCrawler.Api/Startup.cs
using System;
using System.Data;
using System.IO;
using System.Reflection;
using HCrawler.Core;
using HCrawler.DB.Repositories;
using Microsoft.AspNetCore.Builder;
using Microsoft.AspNetCore.Hosting;
using Microsoft.Extensions.Configuration;
using Microsoft.Extensions.DependencyInjection;
using Microsoft.Extensions.Hosting;
using Microsoft.OpenApi.Models;
using Npgsql;

namespace HCrawler.Api
{
    public class Startup
    {
        private readonly IConfiguration _configuration;

        public Startup(IConfiguration configuration)
        {
            _configuration = configuration;
        }

        public void ConfigureServices(IServiceCollection services)
        {
            services.AddControllers();
            services.AddControllersWithViews();
            services.AddHttpClient();

            services.AddScoped<Image.Image>();
            services.AddScoped<IImageRepository, ImageRepository>();
            services.AddScoped<IDownloader, Downloader>();
            
            services.AddHostedService<ConsumeRabbitMQHostedService>();  

            services.AddScoped<IDbConnection>(t =>
            {
                var dbConnection = new NpgsqlConnection(_configuration["ConnectionString"]);
                dbConnection.Open();
                return dbConnection;
            });

            // Register the Swagger generator, defining 1 or more Swagger documents
            services.AddSwaggerGen(c =>
            {
                c.SwaggerDoc("v1",
                    new OpenApiInfo
                    {
                        Title = "Recipe API", Version = "v1", Description = "If you got here you know what it does"
                    });
                var xmlFile = $"{Assembly.GetExecutingAssembly().GetName().Name}.xml";
                var xmlPath = Path.Combine(AppContext.BaseDirectory, xmlFile);
                c.IncludeXmlComments(xmlPath);
            });
        }

        public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
        {
            if (env.IsDevelopment())
            {
                app.UseDeveloperExceptionPage();
            }

            app.UseSwagger();
            app.UseSwaggerUI(c => { c.SwaggerEndpoint("/swagger/v1/swagger.json", "My API V1"); });
            app.UseRouting();
            app.UseStaticFiles();
            app.UseEndpoints(endpoints =>
            {
                endpoints.MapControllers();
                endpoints.MapControllerRoute(
                    name: "default",
                    pattern: "{controller=Home}/{action=Index}/{id?}");
            });
        }
    }
}

File: HCrawler.Api/Startup.fs
namespace HCrawler.Api

open System.Data
open System.Linq
open HCrawler.Consumers
open HCrawler.Core
open HCrawler.Core.Image
open HCrawler.DB.Repositories
open Microsoft.AspNetCore.Builder
open Microsoft.AspNetCore.Hosting
open Microsoft.Extensions.Configuration
open Microsoft.Extensions.DependencyInjection
open Microsoft.Extensions.Hosting
open Npgsql


type Startup(configuration: IConfiguration) =

    // This method gets called by the runtime. Use this method to add services to the container.
    member this.ConfigureServices(services: IServiceCollection) =
        // Add framework services.
        services.AddControllers() |> ignore

        services.AddScoped<IImageRepository, ImageRepository>()
        |> ignore
        services.AddScoped<Image>() |> ignore

        services.AddScoped<IDbConnection>(fun t ->
            let dbConnection =
                new NpgsqlConnection(configuration.["ConnectionString"])

            dbConnection.Open() |> ignore
            dbConnection :> IDbConnection)
        |> ignore


        if (configuration.GetChildren().Any(fun item -> item.Key = "rabbit")) then
            services.AddHostedService<InstagramHostedService>()
            |> ignore
            services.AddHostedService<RedditHostedService>()
            |> ignore

    // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
    member this.Configure(app: IApplicationBuilder, env: IWebHostEnvironment) =
        if (env.IsDevelopment())
        then app.UseDeveloperExceptionPage() |> ignore

        app.UseRouting() |> ignore

        app.UseAuthorization() |> ignore

        app.UseEndpoints(fun endpoints -> endpoints.MapControllers() |> ignore)
        |> ignore

